### DISCUSSION TITLE
Is there something similar to tailwind @apply for new bootstrap utilities

### DISCUSSION THREAD

**Discussion by leomao10** [twbs/bootstrap#31281] [General] [answered]
I am quick exciting that in bootstrap 5 we are going to have utilities class. However, one thing I didn't see in the doco is about how to refactoring the css class in the html. 

You see, applying utilities class is quick an easy, but it does generate a large number of css in the html and create lot of noise. And what is provided by tailwind is [`@apply` directive](https://tailwindcss.com/docs/extracting-components/#extracting-css-components-with-apply), so eventually we would ends up with something like:

```html
<button class="btn-blue">
  Button
</button>
```
compare to:
```html
<button class="bg-blue-500 text-white font-bold py-2 px-4 rounded">
  Button
</button>
```

Is there anything planned to build in bootstrap 5's roadmap?

**Comment by mdo** [comment, marked as answer]
> Is there anything planned to build in bootstrap 5's roadmap?

No, nothing planned like this.

**Reply by leomao10** [reply]
Thanks for the quick response, so is there a suggested way to do refactoring after? maybe mixins provided by bootstrap  or simply use `@extend` 🤔

**Reply by acrodrig** [reply]
I have the same question(s):

- What is the "bootstrap approved" way of simulating `@apply?` 
- Is it considered bad form? 
- Should we use `@extend` instead?

Thanks!

**Reply by ghost** [reply]
Can't we make a .scss file and apply those?

**Reply by martinbean** [reply]
```scss
.your-selector {
  @extend .m-3;
  // and any other utility class you want
}
```
